protected/controllers/AdminSampleController.php

```diff
@@ -205,189 +189,188 @@ public function actionChoose() {
-                $data = explode('=', $sAttr);
-                if (count($data) == 2) {
-                    
-                    // Get attribute, if not exist we create
-                    $attribute = Attribute::model()->findByAttributes(array('structured_comment_name' => trim($data[0])));
+            $sampleAttribute = new SampleAttribute();
+            $sampleAttribute->sample_id = $model->id;
+
+            foreach (explode('",', $model->attributesList) as $attributes) {
+                $attributes = str_replace('"', '', $attributes);
+                $attributeData = explode('=', $attributes);
+                if (count($attributeData) == 2) {
+                    // Get attribute model
+                    $attribute = Attribute::model()->findByAttributes(array('structured_comment_name' => trim($attributeData[0])));
                     if (!$attribute) {
-                        $attribute = new Attribute();
-                        $attribute->structured_comment_name = trim($data[0]);
-                        $attribute->attribute_name = str_replace('_', ' ', trim($data[0]));
-                        $attribute->definition = str_replace('_', ' ', trim($data[0]));
-                        $attribute->save();
+                        $model->addError('error', 'Attribute name for the input ' . $attributeData[0]. "=" . $attributeData[1] .' is not valid - please select a valid attribute name!');
                     }
-                    
                     // Let's save the new sample attribute
-                    $sampleAttribute = clone $sampleAttr;
-                    $sampleAttribute->value = trim($data[1]);
+                    $sampleAttribute = clone $sampleAttribute;
+                    $sampleAttribute->value = trim($attributeData[1]);
                     $sampleAttribute->attribute_id = $attribute->id;
-                    $sampleAttribute->save();
+                    if (!$sampleAttribute->save()) {
+                        foreach ($sampleAttribute->getErrors() as $errors) {
+                            foreach ($errors as $errorMessage) {
+                                $model->addError('error', $errorMessage);
+                            }
+                        }
+                    }
                 }
             }
         }
     }
+
+    /**
+     * @param $tax_id
+     * @param $model
+     * @return array|CActiveRecord|mixed|Species|null
+     */
+    private function getSpecies($tax_id, $model)
+    {
+        $species = Species::model()->findByAttributes(array('tax_id' => $tax_id));
+        if (!$species) {
+            $model->addError('error', 'Species id ' . $tax_id . ' is not found!');
+        }
+        $model->species_id = $species->id;
+        $model->attributesList = $_POST['Sample']['attributesList'];
+        $this->updateSampleAttributes($model);
+        if (!$model->errors) {
+            $model->save();
+            $this->redirect(array('view', 'id' => $model->id));
+        }
```

Comment:
**thought**: Saving the model here feels out of place in this function. Should it be in the `actionCreate` function above?

Reply:
yes, agreed, see my comment above about getters and setters